generic-contracts/src/main/java/com/scalar/dl/genericcontracts/table/v1_0_0/Select.java

```diff
@@ -18,30 +27,128 @@ public JsonNode invoke(
       Ledger<JsonNode> ledger, JsonNode arguments, @Nullable JsonNode properties) {
 
     // Check the arguments
-    if ((arguments.size() != 2 && arguments.size() != 3)
-        || !arguments.has(Constants.QUERY_TABLE)
-        || !arguments.get(Constants.QUERY_TABLE).isTextual()
-        || !arguments.has(Constants.QUERY_CONDITIONS)
-        || !arguments.get(Constants.QUERY_CONDITIONS).isArray()) {
-      throw new ContractContextException(Constants.INVALID_QUERY_FORMAT);
-    }
+    validateQuery(arguments);
 
-    // Check the format of each condition
-    JsonNode conditions = arguments.get(Constants.QUERY_CONDITIONS);
-    validateConditions(conditions);
+    // Prepare conditions for each table
+    JsonNode firstTable = arguments.get(Constants.QUERY_TABLE);
```

Comment:
```suggestion
    JsonNode leftmostTable = arguments.get(Constants.QUERY_TABLE);
```
I feel `first` probably sounds a bit unclear to readers.